### PR TITLE
Fixed 'Bug 55329 - [regression] IntelliSense not working with

### PR DESCRIPTION
**Pull request by mkrueger**
excluding conditional ItemGroup'

This patch prevents the document in beeing in state Project != null &&
AnalysisDocument == null.

### CODE REVIEW

**Comment by Therzok** [comment]
Best seen in https://github.com/mono/monodevelop/pull/2217/files?w=1

**Comment by slluis** [comment]
What will happen if a file is opened while the type system is still being loaded and there is yet no type system document for the file? won't then the file be assigned an ad-hoc project? will it get the real roslyn project when type information is finally loaded for the file?

**Comment by mkrueger** [comment]
Worked for me in the monodevelop solution. However it may be because the documents are created very quickly but the parsing/loading lags behind. Anyways that patch fixed it.

During reload files could be included in the project via item group condition change.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs [hunk 1]
```diff
@@ -846,54 +845,56 @@ Task EnsureAnalysisDocumentIsOpen ()
...
 				RoslynWorkspace = TypeSystemService.GetWorkspace (this.Project.ParentSolution);
+				if (RoslynWorkspace == null) // Solution not loaded yet
+					return Task.CompletedTask;
 				SubscribeRoslynWorkspace ();
 				analysisDocument = FileName != null ? TypeSystemService.GetDocumentId (this.Project, this.FileName) : null;
 				if (analysisDocument != null) {
 					TypeSystemService.InformDocumentOpen (analysisDocument, Editor);
+					return Task.CompletedTask;
 				}
-			} else {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui/Document.cs]: Either the removal of the else clause without adding a return Task.CompletedTask in the previous if condition or the change from `IsAdHocProject` to `UnloadAdhocProject` turned this into an infinite loop.

**Pull request merged**